core/services/ardupilot_manager/typedefs.py

```diff
@@ -163,3 +166,38 @@ class EndpointType(str, Enum):
     TCPServer = "tcpin"
     TCPClient = "tcpout"
     Serial = "serial"
+
+
+class Serial(BaseModel):
+    """Simplified representation of linux serial port configurations,
+    gets transformed into command line arguments such as
+    --serial1 /dev/ttyS0
+    --serial3 /dev/ttyAMA1
+    --serial4 /dev/ttyAMA2
+    --serial5 /dev/ttyAMA3
+    """
+
+    port: str
+    endpoint: str
+
+    @validator("port")
+    def valid_number(cls: Any, value: str) -> str:
+        if value in "BCDEFGH" and len(value) == 1:
+            return value
+        raise ValueError(f"Invalid serial port: {value}. These must be between B and H. A is reserved.")
+
+    @validator("endpoint")
+    def valid_endpoint(cls: Any, value: str) -> str:
+        if Path(value).exists():
+            return value
+        if re.compile(r"tcp:\d*:wait$").match(value):
+            return value
+        matches = re.compile(r"(tcpclient|udpclient|tcpin|udpin):(?P<ip>(\d*\.){3}\d+):(?P<port>\d*)$").match(value)
```

Comment:
Should we also support `serial` ? `serial:/dev/ttyUSB2:115200`

Reply:
is there an advantage in doing it? it sounds like additional work with not much to gain.

Reply:
Well, it's a valid endpoint, just saying